src/parser.rs

```diff
@@ -1495,48 +1497,43 @@ pub fn parse_function(
 
                 let mut return_type = ParsedType::Empty;
 
-                let mut fat_arrow_expr = None;
-
-                if *index + 2 < tokens.len() {
-                    match &tokens[*index].contents {
-                        TokenContents::FatArrow => {
-                            *index += 1;
-
-                            let (expr, err) = parse_expression(
-                                tokens,
-                                index,
-                                ExpressionKind::ExpressionWithoutAssignment,
-                            );
-                            return_type = ParsedType::Empty;
-                            fat_arrow_expr = Some(expr);
-                            error = error.or(err);
-
-                            *index += 1;
-                        }
-                        TokenContents::Minus => {
-                            *index += 1;
-
-                            match &tokens[*index].contents {
-                                TokenContents::GreaterThan => {
-                                    *index += 1;
-
-                                    let (ret_type, err) = parse_typename(tokens, index);
-                                    return_type = ret_type;
-                                    error = error.or(err);
-                                }
-                                _ => {
-                                    trace!("ERROR: expected ->");
-
-                                    error = error.or(Some(JaktError::ParserError(
-                                        "expected ->".to_string(),
-                                        tokens[*index - 1].span,
-                                    )));
-                                }
-                            }
-                        }
-                        _ => {}
+                // accept return type specification with '->'
+                let return_type_span = if tokens[*index].contents == TokenContents::Minus {
+                    *index += 1;
+                    if tokens[*index].contents != TokenContents::GreaterThan {
+                        error = error.or(Some(JaktError::ParserError(
+                            "expected ->".to_string(),
+                            tokens[*index - 1].span,
+                        )));
                     }
-                }
+                    let start = *index;
+                    let file_id = tokens[*index].span.file_id;
+                    *index += 1;
+                    let (ret_type, err) = parse_typename(tokens, index);
+                    return_type = ret_type;
+                    error = error.or(err);
+                    Some(Span {
+                        file_id,
+                        start,
+                        end: tokens[*index - 1].span.end,
+                    })
+                } else {
+                    None
+                };
+
+                // accept an (optional)  fat arrow
```

Comment:
double space in comment here

Reply:
That code goes through automatic formatter. I can't enforce that manually.

Reply:
Nevermind, I could enforce that manually.